readline.c

```diff
@@ -67,11 +78,12 @@ readline_last_from_rl(readline_t * rl) {
   ret = (char*) malloc(len);
 
   if (ret == NULL) {
+    return NULL;
```

Comment:
Assert's aren't for checking errors. Assert's indicates assumption of a code. On other words, an assertion specifies that a program/library satisfies certain conditions at particular points in its execution.
Assert macro are only active when the source code is compiled with debug flags. Release code, have assert macro disabled. So when the library is compiled on debug mode, the user or the developer of the lib, could see if a problem comes of a bad usaged of the lib or of a real bug on the lib. For example, if a user code pass a bad readline_t where buffer is NULL, the user code is wrong, and the debug version of the lib would fail showing that.

More info : http://ptolemy.eecs.berkeley.edu/~johnr/tutorials/assertions.html